core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3105,4 +3270,153 @@ public Long getValue() {
+    private boolean mSuccess;
+    private String mCommand;
+    private AlluxioURI mSrcPath;
+    private AlluxioURI mDstPath;
+    private String mUser;
+    private String mIp;
+    private Inode mSrcInode;
+
+    @Override
+    public MasterAuditContext setAllowed(boolean allowed) {
+      mAllowed = allowed;
+      return this;
+    }
+
+    @Override
+    public MasterAuditContext setStatus(boolean success) {
+      mSuccess = success;
+      return this;
+    }
+
+    /**
+     * Sets mCommand field.
+     *
+     * @param command the command associated with this {@link alluxio.master.Master}
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setCommand(String command) {
+      mCommand = command;
+      return this;
+    }
+
+    /**
+     * Sets mSrcPath field.
+     *
+     * @param srcPath the source path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcPath(AlluxioURI srcPath) {
+      mSrcPath = srcPath;
+      return this;
+    }
+
+    /**
+     * Sets mDstPath field.
+     *
+     * @param dstPath the destination path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setDstPath(AlluxioURI dstPath) {
+      mDstPath = dstPath;
+      return this;
+    }
+
+    /**
+     * Sets mUser field.
+     *
+     * @param user the client user name of the authenticated client user of this thread
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setUser(String user) {
+      mUser = user;
+      return this;
+    }
+
+    /**
+     * Sets mIp field.
+     *
+     * @param ip the IP of the client
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setIp(String ip) {
+      mIp = ip;
+      return this;
+    }
+
+    /**
+     * Sets mSrcInode field.
+     *
+     * @param srcInode the source inode of this operation
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcInode(Inode srcInode) {
+      mSrcInode = srcInode;
+      return this;
+    }
+
+    /**
+     * Constructor of {@link MasterAuditContext}.
+     *
+     * @param asyncAuditLogWriter
+     */
+    private MasterAuditContext(AsyncUserAccessAuditLogWriter asyncAuditLogWriter) {
+      mAsyncAuditLogWriter = asyncAuditLogWriter;
+      mAllowed = true;
+    }
+
+    @Override
+    public void close() {
+      if (mAsyncAuditLogWriter == null) {
+        return;
+      }
+      mAsyncAuditLogWriter.append(this);
+    }
+
+    @Override
+    public String toString() {
+      if (mSrcInode != null) {
+        return String.format("allowed=%b\tuser=%s\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\t"
+                + "perm=%s:%s:%d\tsuccess=%b",
+            mAllowed, mUser, mIp, mCommand, mSrcPath, mDstPath, mSrcInode.getOwner(),
+            mSrcInode.getGroup(), mSrcInode.getMode(), mSuccess);
```

Comment:
`mSrcInode.getMode()` is a short, not even in Octal which can be confusing. 
if you want to show the rwx strings, checks out `alluxio.security.authorization.Mode.Bits#fromShort()`